crypto/rand/rand_lib.c

```diff
@@ -36,15 +112,33 @@
 # include "internal/e_os.h"
 # include "internal/property.h"
 
+/*
+ * The default name for the random provider.
+ * This ensures that the FIPS provider will supply libcrypto's random byte
+ * requirements.
+ */
+static const char random_provider_fips_name[] = "fips";
+
+static int set_random_provider_name(RAND_GLOBAL *dgbl, const char *name)
+{
+    if (dgbl->random_provider_name != NULL
+            && strcmp(dgbl->random_provider_name, name) == 0)
```

Comment:
I think provider name comparisons should be case insensitive - but that is a separate outstanding discussion elsewhere ...

Reply:
done